frontend/packages/noobaa-storage-plugin/src/components/capacity-card/capacity-card.tsx

```diff
@@ -0,0 +1,244 @@
+  // limit the numbers by 'sizeLimit', rest of the items should be put into 'Others'
+  const myDataExtra = myDataSorted.slice(sizeLimit, myDataSorted.length);
+  // first N items (where N=sizeLimit)
+  myDataSorted = myDataSorted.slice(0, sizeLimit);
+  // if there are any extra / other items
+  if (myDataExtra.length > 0) {
+    const otherData: IChartData = { x: 'Others', y: 0 };
+    myDataExtra.forEach((eachData) => {
+      otherData.y += eachData.y;
+    });
+    myDataSorted.push(otherData);
+  }
+  return myDataSorted;
+}
+
+// returns the IChartData in percentage, and the total
+function convertToPercentage(data: IChartData[]): [IChartData[], number] {
+  const retData = new Array<IChartData>();
+  let total = 0;
+  // d: will be containing value
+  for (const d of data) {
+    total += d.y;
+    retData.push({ x: d.x, y: d.y });
+  }
+  // i: will be containing the index
+  Object.keys(retData).forEach((i) => (retData[i].y = (retData[i].y / total) * 100));
+  return [retData, total];
+}
+
+const ObjectDashboardCapacityCard: React.FC<DashboardItemProps> = ({
+  watchPrometheus,
+  stopWatchPrometheusQuery,
+  prometheusResults,
+}) => {
+  React.useEffect(() => {
+    Object.keys(QueryType).forEach((k) =>
+      watchPrometheus(getNooBaaCapacityUsageQuery(QueryType[k])),
+    );
+    return () => {
+      Object.keys(QueryType).forEach((k) => {
+        stopWatchPrometheusQuery(getNooBaaCapacityUsageQuery(QueryType[k]));
+      });
+    };
+  }, [watchPrometheus, stopWatchPrometheusQuery]);
+
+  const cuProjectsResult = prometheusResults.getIn([
+    getNooBaaCapacityUsageQuery(QueryType.Project),
+    'result',
+  ]);
+  const cuBucketClassResult = prometheusResults.getIn([
+    getNooBaaCapacityUsageQuery(QueryType.BucketClass),
+    'result',
+  ]);
+
+  const isLoading = !(cuProjectsResult && cuBucketClassResult);
+
+  const cuProjectData = getCapacityUsage(cuProjectsResult, QueryType.Project);
+  const [cuProjectDataInPercentage, cuProjectTotal] = convertToPercentage(cuProjectData);
+  const cuProjectLegends = getCapacityUsageLegends(cuProjectData);
+
+  const cuBucketClassData = getCapacityUsage(cuBucketClassResult, QueryType.BucketClass);
+  const [cuBucketClassDataInPercentage, cuBucketClassTotal] = convertToPercentage(
+    cuBucketClassData,
+  );
+  const cuBucketClassLegends = getCapacityUsageLegends(cuBucketClassData);
+
+  const [capacityUsageType, setCapacityUsageType] = React.useState(QueryType.Project);
+
+  const chartDataMap = new Map<QueryType, MetricsData>();
+
+  chartDataMap.set(QueryType.Project, {
+    chartData: cuProjectDataInPercentage,
+    legendData: cuProjectLegends,
+    total: cuProjectTotal,
+  });
+  chartDataMap.set(QueryType.BucketClass, {
+    chartData: cuBucketClassDataInPercentage,
+    legendData: cuBucketClassLegends,
+    total: cuBucketClassTotal,
+  });
+
+  return (
+    <DashboardCard>
+      <DashboardCardHeader>
+        <DashboardCardTitle>
+          <span>Capacity Usage</span>
```

Comment:
does not have to be wrapped in span
```suggestion
          Capacity Usage
```

Reply:
Acknowledged...